sycl/include/sycl/ext/oneapi/bfloat16.hpp

```diff
@@ -97,9 +121,13 @@ class bfloat16 {
   friend bfloat16 operator-(bfloat16 &lhs) {
 #if defined(__SYCL_DEVICE_ONLY__)
 #if defined(__NVPTX__)
+#if (__CUDA_ARCH__ >= 800)
     return from_bits(__nvvm_neg_bf16(lhs.value));
 #else
-    return bfloat16{-__spirv_ConvertBF16ToFINTEL(lhs.value)};
+    return -to_float(lhs.value);
+#endif
+#else
+    return bfloat16{-__devicelib_ConvertBF16ToFINTEL(lhs.value)};
 #endif
 #else
     (void)lhs;
```

Comment:
The line just below is:

```
throw exception{errc::feature_not_supported,
                    "Bfloat16 unary minus is not supported on host device"};
```

Why can't we implement unary minus on the host?  All the other bfloat16 operations are implemented for both host and device, so this is the only one that is not already implemented on the host.  It seems like the implementation is as simple as:

```
return -to_float(lhs.value);
```

If we did this, we could update the extension specification to say that the `bfloat16` type is available in both host and device code.  Actually, the spec isn't very clear about this currently, so a reader might already assume that it is available in host code.